### ISSUE TITLE
[Migration] Token will not issues unregistered scopes

### ISSUE THREAD

**Issue by sahandilshan** [wso2/product-is#18548]
**Describe the issue:**
Once the https://github.com/wso2/product-is/issues/17551 improvements got onboarded
1. We will no longer allow the use of random scopes
2. If the user needs random scopes, they need to add them as allowed scopes
3. Above configured scopes will not show on the consent page

In this PR https://github.com/wso2/carbon-identity-framework/pull/5284, we have mentioned how users can configure and customize this feature. If needed, the user can also disable this feature

**Comment by sahandilshan** [comment]
In the system's default setup, tokens are created with two types of scopes: those that are officially registered and those that fit the ^device_.* regex pattern (like device_1, device_2, etc.). Unregistered scopes that match this pattern are automatically included in the token. However, it's key to understand that these unregistered scopes do not show up on the consent page. This means when a user requests both standard system-registered scopes and a scope that follows the ^device_.* pattern, the consent page will only display the system-registered scopes. Despite this, the final token issued will contain both the system-registered scopes and any scopes that align with the ^device_.* regex pattern.

* **Disabling the Default Scope Inclusion:**
If the user wants to disable this behaviour, they can do it by adding the following configurations into `deployment.toml`
```
[oauth]
drop_unregistered_scopes = "false"
allowed_scopes = []
```
This change will disable the above mentioned behavior and issues a token with any random scope

* **Introducing Custom Scope Patterns**
Users can also define their own custom scope patterns. To do this, they need to modify the deployment.toml file as shown below
```
[oauth]
allowed_scopes = ["^device_.*", "any_custom_regex_pattern"]
```

With this configuration, the system will recognize both the default regex pattern `(device_.*)` and any user-defined custom regex patterns as valid for inclusion in tokens. If users prefer to use only their custom patterns, excluding the default, they can simply remove the `^device_.*` pattern from the `allowed_scopes` configuration and include only their specified patterns and scopes. This setup allows for the inclusion of both regex patterns and plain string text in the `allowed_scopes` configuration.

In summary, these configurations offer users flexibility in managing which scopes are included in tokens, be it through the system's default settings or through customized configurations defined in the `deployment.toml` file.
